client/src/components/AuthenticationForm.js

```diff
@@ -45,35 +46,111 @@ const form = {
     switchFormQuestion: 'Dont have an account?',
     anchor: '/signup',
     linkTo: 'SignUp',
-    switchFormAnswer: 'Sign up!',
+    switchFormAnswer: 'Signup!',
   },
-  signUp: {
-    buttonText: 'Sign up',
+  signup: {
+    buttonText: 'Signup',
     switchFormQuestion: 'Already have an account?',
     anchor: '/login',
     linkTo: 'Login',
     switchFormAnswer: 'Login!',
   },
 };
 
-function AuthenticationForm({ type, valueUsername, valuePassword }) {
+function AuthenticationForm({ type }) {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const history = useHistory();
+
+  async function loginUser() {
+    try {
+      const response = await fetch(`http://localhost:8080/users/login`, {
```

Comment:
This will not work on Heroku (or on any other deployment), because localhost:8080 depends on your local development environment.

That's why you should have a proxy configured.
https://create-react-app.dev/docs/proxying-api-requests-in-development/

```suggestion
      const response = await fetch(`/api/users/login`, {
```

Make sure to add `/api` on your server-side too. Otherwise, conflicts with the routes defined in react-router might be possible.

This issue happens multiple times.